src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/EndOffenderProgramProfileRequest.kt

```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.activities
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.swagger.v3.oas.annotations.media.Schema
+import org.hibernate.validator.constraints.Length
+import java.time.LocalDate
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(description = "Course activity creation request")
+data class EndOffenderProgramProfileRequest(
+  @Schema(description = "Activity end date", required = true, example = "2022-08-12")
+  val endDate: LocalDate,
+
+  @Schema(description = "Activity end reason (from domain PS_END_RSN)", example = "REL")
```

Comment:
It'll be interesting to see how this is modelled in the Activities UI - I guess we'll be mapping their reasons to Nomis reasons?

Reply:
yes, quite possibly. We wont know for a while yet though i expect